### PR TITLE
Added gold-a647 mic comparisons to  mic analysis

### PR DESCRIPTION
**Pull request by MattsonCam**
This pr adds GOLD-A647 MIC comparisons to the MIC analysis. Any constructive feedback is welcome when reviewing this pr.

### CODE REVIEW

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py [hunk 1]
```diff
@@ -251,12 +247,12 @@
...
     plt.yticks(fontsize=12)
 
-    plt.title(f"Distributions of Maximal Information Coeficient (MIC) Scores\nBetween DAPI and {stain} Features per Well", fontsize=16)
+    plt.title(f"Distributions of Maximal Information Coeficient (MIC) Scores\nBetween {comparison_stains[0]} and {comparison_stains[1]} Features per Well", fontsize=16)
 
     plt.xlim(0, 1)
     plt.ylim(0.0, 90.0)
 
-    plt.savefig(distribution_figures_path / f"mic_distributions_dapi_{stain.lower()}_wells.png")
+    plt.savefig(distribution_figures_path / f"mic_distributions_{comparison_stains[0].lower()}_{comparison_stains[1].lower()}_wells.png")
```
Comment (MikeLippincott) [inline 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py:255]: Nit pick here, consider both resolving the path and making a pathlib path.

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py [hunk 2]
```diff
@@ -234,11 +229,12 @@
...
-for stain in speckle_stains:
+stains = {"DAPI", "GOLD", "A647"}
+
+for stain in stains:
 
-    pattern = f"DAPI|{re.escape(stain)}"
-    stain_micdfs = micdfs.loc[micdfs["Metadata_Stain__posthoc_group0"].str.contains(pattern, regex=True) & micdfs["Metadata_Stain__posthoc_group1"].str.contains(pattern, regex=True)]
-    print(stain_micdfs)
+    comparison_stains = list(stains - {stain})
+    stain_micdfs = micdfs.loc[~(micdfs["Metadata_Stain__posthoc_group0"].str.contains(stain, regex=True) | micdfs["Metadata_Stain__posthoc_group1"].str.contains(stain, regex=True))]
```
Comment (MikeLippincott) [inline 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py]: consider changing naming here for increased readability. stain vs stains is hard for me to differentiate. Consider maybe using `list_of_stains`

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py [hunk 3]
```diff
@@ -170,33 +169,29 @@
...
+        mic_comparator = ShuffledMIC()
 
-        speckle_staindf = staindf.loc[staindf["Metadata_Stain"] != stain]
+    comparer = PairwiseCompare(
+        _df=staindf.copy(),
+        _comparator=mic_comparator,
+        _antehoc_group_cols=["Metadata_Plate", "Metadata_Well", "Metadata_Condition"],
+        _posthoc_group_cols=["Metadata_Stain"],
+        _feat_cols=feat_cols,
+    )
```
Comment (MikeLippincott) [inline 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py:188]: This is sharp, Nice!

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py [hunk 4]
```diff
@@ -2,17 +2,16 @@
 # coding: utf-8
 
 # # Evaluate Well Stain Relationship Strength
-# This analysis compares Maximal Information Coefficient (MIC) scores between the same DAPI and nuclear speckle stain (A647 or GOLD) aggregated well features per well.
+# This analysis compares Maximal Information Coefficient (MIC) scores between DAPI and nuclear speckle stain (A647 and GOLD) aggregated well features per well.
 # Distributions of these MIC scores are visualized, between zero and one, where one indicates a perfect relationship and zero indicates no relationship.
-# From this analysis we can understand the relationships between the DAPI and nuclear speckle stains per cell population (Treatment, Well Position, Plate, etc...)
+# From this analysis we can understand the relationships between these stains per cell population (Treatment, Well Position, Plate, etc...)
```
Comment (MikeLippincott) [inline 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py:7]: I love these, this sums up this script very well!

**Review by MikeLippincott — APPROVED** [review]
LGTM, Nice PR!

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py (hunk 1 above)
Comment (MattsonCam) [reply 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py:255]: I don't see a benefit to doing this. I might just keep it as is

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py (hunk 2 above)
Comment (MattsonCam) [reply 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py]: I made `stain` more descriptive

File: 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py (hunk 4 above)
Comment (MattsonCam) [reply 0.data_analysis_and_processing/mic_analysis/nbconverted/well_sirna_mic_analysis.py:7]: Thanks!

**Comment by MattsonCam** [comment]
Thanks @MikeLippincott, merging now!

**Pull request merged**